### ISSUE TITLE
Why not call write directly?

### ISSUE THREAD

**Issue by ioquatix** [havenwood/roda-sse#1]
https://github.com/havenwood/roda-sse/blob/a7f8b531a6a065d7e5f12d3f52f4a667a24df247/lib/roda/plugins/sse.rb#L31

**Comment by havenwood** [comment]
I was originally calling `write` directly and handling closing the stream at the Roda app level. As a feature, I wanted to have `r.sse do |stream|` blocks ensure `stream.close` but that requires either having the plugin author ensure their block returns `stream`, which seemed onerous, or some way to capture the `stream` right before it's called.

Roda doesn't really give Rack 3 stream access, and I couldn't find a way to reliably get to the connection from Roda to close it manually in the plugin. I borrowed the `Stream` abstraction from the Roda streaming plugin, but felt it could handle both intercepting the `stream` and wrapping the write with Async.

First thought I'd do Async stuff in app land, but found I needed a Barrier to loop (whether it's actually needed or a skill issue on my side, I'm not sure). I encountered a deluge of Tasks that weren't closing on disconnection errors. My thinking in adding the barrier was to ensure that the current Task and all other pending ones for the stream are closed. With the barrier, I just see the single error percolate up.

I very much appreciate you taking a look. Any feedback or correction is super welcome. Thank you!
